lib/iris/coords.py

```diff
@@ -1097,7 +1117,10 @@ def contiguous_bounds(self):
                           'contiguous bounds.'.format(self.name()))
             bounds = self._guess_bounds()
         else:
-            self._sanity_check_contiguous()
+            if not self.is_contiguous():
+                raise ValueError('Invalid operation. Bounds of coord {!r} '
+                                 'are not contiguous.'.format(self.name()))
+
```

Comment:
Nice catch.  Much better.

Reply:
Unfortunately this is a breaking change so I am going to have to put this back